cli/internal/nodeprovider/local/createNodeIfNotExists.go

```diff
@@ -32,63 +33,54 @@ func (np nodeProvider) CreateNodeIfNotExists(ctx context.Context) (node.Node, er
 		return nil, err
 	}
 
-	nodeCmd := exec.Command(
-		pathToOpctlBin,
+	cmdName := pathToOpctlBin
+	cmdArgs := []string{
 		"--data-dir",
-		np.dataDir.Path(),
-		"--listen-address",
-		np.config.ListenAddress,
+		np.config.DataDir,
+		"--api-listen-address",
+		np.config.APIListenAddress,
 		"--container-runtime",
 		np.config.ContainerRuntime,
 		"node",
 		"create",
+	}
+
+	if os.Geteuid() != 0 {
```

Comment:
opctl must run as root now in order to manage dns and network

Reply:
Have we considered other options like only running the dns configuration steps as root in an isolated way. if a user doesnt want to run as root are they unable to use this version?

Reply:
It's effectively already in an isolated way in the sense It's only the node/daemon that's required to run as root. This is similar to how other tools do it e.g. docker for Mac takes this same approach.